src/Utilities/sparse_matrix.jl

```diff
@@ -116,61 +118,71 @@ function final_touch(A::MutableSparseMatrixCSC)
     for i in length(A.colptr):-1:2
         A.colptr[i] = _shift(A.colptr[i-1], ZeroBasedIndexing(), A.indexing)
     end
-    return A.colptr[1] = _first_index(A.indexing)
+    A.colptr[1] = _first_index(A.indexing)
+    return
 end
 
 _variable(term::MOI.ScalarAffineTerm) = term.variable
 _variable(term::MOI.VectorAffineTerm) = _variable(term.scalar_term)
-function _allocate_terms(colptr, index_map, terms)
-    for term in terms
-        colptr[index_map[_variable(term)].value+1] += 1
-    end
-end
```

Comment:
@blegat what was the motivation to have these `_allocate_terms` function barriers?

Reply:
It dates back to https://github.com/jump-dev/SCS.jl/pull/192 where I benchmarked the two approaches and the function barrier seemed to be faster even if I didn't really get why.
It's maybe not the case anymore in Julia v1.6 or maybe I was mistaken when writing this.
We can revisit later. An easier speedup would be obtained by using `@inbounds` I guess.
The good thing with MatrixOfConstraints is that we will be able to spend time optimizing these pieces at once in MOI and it will affect all solver's `copy_to` performance so it might start being worth playing with `@inbounds`, etc...

Reply:
Given the massive amount of new complexity we're introducing, I'd rather that we merged a simplified version first, got it working and debugged, then wrote the various solvers, and only then revisited performance implications with benchmarks.

Reply:
Yes, definitely